### DISCUSSION TITLE
Workflow and steps for Local to Production enviroments

### DISCUSSION THREAD

**Discussion by WetDry** [directus/directus#17009] [Archive]
Hello guys, i'm a Laravel dev and i want to start making some project with Directus. I've a trouble about local-staging-production env.
For example, using Laravel i'm able to make some changes to my database in my Local env, then move to staging or production, run migrations and everything is aligned, without touch DB and more important the data.
With Directus if i'm working on a Local project and want to move in Production i can dump db and put on Prod Db server.
But if i make some changes, more development in my Local env and i want to deploy that changes in Production without make down server and without loss data? 
I hope that Dump or more in general working direct on DB is not the only solution... I'm using managed DB because i don't want to take care of database it would make me laugh if for every minimun changes i need to touch directly database it just makes useless Directus itself

PS: PS i've serached on google about my request but i can't find any documentation about snapsot or YAML file, seem removed from official documentation  https://docs.directus.io/reference/cli.html

Thanks a lot for support and information about my request

**Comment by danielband** [comment]
Hi, I think what your looking for is [Command Line Interface](https://docs.directus.io/self-hosted/cli.html#snapshot-the-data-model). You have to be careful though, what kind of changes will apply, or you might have data loss. Directus **does not** magically save your data for you. So pay attention before deleting things. Depending on your situation (defining indices, renaming stuff, etc.), you should define [custom migrations](https://docs.directus.io/extensions/migrations.html), that get executed when `directus database migrate` is run.

**Comment by EdouardDem** [comment]
Hi, I've developped a CLI for this purpose: https://github.com/tractr/directus-sync
It synchronizes schema, dashboards, flows, folders, operations, panels, permissions, presets, roles, settings, translations, and webhooks across multiple Directus instances.
I have also written a tutorial : https://medium.com/tractr/streamlining-directus-workflows-with-directus-sync-a-comprehensive-guide-baf33b9cc399
Hope this helps
